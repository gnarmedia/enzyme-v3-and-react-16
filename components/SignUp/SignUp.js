import React, { Component } from 'react'

import Input from '../Input/Input'
import Button from '../Button/Button'

class SignUp extends Component {
  handleSubmit = (event) => {
    event.preventDefault();

    if (!event.target.checkValidity()) {
      // form is invalid! so we do nothing
      console.log('invalid!');
      return;
    }

    console.log('valid!');
  }

  render() {
    return (
      <section className="is-fullheight">
        <div className="hero-body">
          <div className="container has-text-centered">
            <div className="column is-4 is-offset-4">
              <h3 className="title has-text-grey">Sign Up</h3>
              <div className="box">
                <figure className="avatar">
                  <img src="https://placehold.it/128x128" alt="placeholder" />
                </figure>
                <form noValidate onSubmit={this.handleSubmit}>
                  <Input name="email" placeholder="Your Email address" type="email" required />
                  <Input name="password" placeholder="Your password" type="password" required />
                  <Input name="passwordConfirm" placeholder="Confirm password" type="password" required />
                  <Button>Sign Up</Button>
                </form>
              </div>
            </div>
          </div>
        </div>
      </section>
    )
  }
}

export default SignUp