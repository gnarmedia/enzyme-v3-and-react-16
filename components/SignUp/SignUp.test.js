import React from 'react'
import { render } from 'enzyme'

import SignUp from './SignUp'

describe('index', () => {
  it('should render the index page', () => {
    const wrapper = render(<SignUp />)

    expect(wrapper).toMatchSnapshot()
  })
})