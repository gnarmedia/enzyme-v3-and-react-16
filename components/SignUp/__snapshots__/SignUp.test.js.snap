// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index should render the index page 1`] = `
<section
  class="is-fullheight"
>
  <div
    class="hero-body"
  >
    <div
      class="container has-text-centered"
    >
      <div
        class="column is-4 is-offset-4"
      >
        <h3
          class="title has-text-grey"
        >
          Sign Up
        </h3>
        <div
          class="box"
        >
          <figure
            class="avatar"
          >
            <img
              alt="placeholder"
              src="https://placehold.it/128x128"
            />
          </figure>
          <form
            novalidate=""
          >
            <div
              class="field"
            >
              <div
                class="control"
              >
                <input
                  class="input is-large"
                  name="email"
                  placeholder="Your Email address"
                  required=""
                  type="email"
                />
              </div>
            </div>
            <div
              class="field"
            >
              <div
                class="control"
              >
                <input
                  class="input is-large"
                  name="password"
                  placeholder="Your password"
                  required=""
                  type="password"
                />
              </div>
            </div>
            <div
              class="field"
            >
              <div
                class="control"
              >
                <input
                  class="input is-large"
                  name="passwordConfirm"
                  placeholder="Confirm password"
                  required=""
                  type="password"
                />
              </div>
            </div>
            <button
              class="button is-block is-info is-large is-fullwidth"
              type="button"
            >
              Sign Up
            </button>
          </form>
        </div>
      </div>
    </div>
  </div>
</section>
`;
